Common/Data/Encoding/Utf8.cpp

```diff
@@ -628,12 +570,13 @@ static size_t ConvertUTF8ToWStringInternal(wchar_t *dest, size_t destSize, const
 	return dest - orig;
 }
 
-std::wstring ConvertUTF8ToWString(const std::string &source) {
+std::wstring ConvertUTF8ToWString(std::string_view source) {
 	std::wstring dst;
-	// utf-8 won't be less bytes than there are characters.  But need +1 for terminator.
-	dst.resize(source.size() + 1, 0);
-	size_t realLen = ConvertUTF8ToWStringInternal(&dst[0], source.size() + 1, source);
+	// utf-8 won't be less bytes than there are characters.
+	dst.resize(source.size(), 0);
+	size_t realLen = ConvertUTF8ToWStringInternal(&dst[0], source.size(), source);
 	dst.resize(realLen);
+	dst[realLen] = 0;
```

Comment:
Something doesn't look right here.  You're setting `[realLen]` to 0, but you've intentionally not added one for the terminator.  In practice, this may be fine - I think std::wstring will always have a null there, but it seems unsafe?

Suppose the string were just ASCII codepoints, let's say 5 letters: hello.  `dst.resize(5);` and `realLen` is returned as 5 (no terminator written because it hit destSize), but then you'll write to `dst[5]`.

I guess this write isn't needed maybe?  But not sure if any callers somehow depend on it, as it was there before.

-[Unknown]

Reply:
Uh yeah, I screwed this one up it seems. After resizing a string, there is indeed no need to write the terminator and I believe that doing so is undefined though will work in practice (but may be caught by debugging tools). Will fix.